compiler/src/iree/compiler/Codegen/Common/OptimizeTensorInsertExtractSlices.cpp

```diff
@@ -240,8 +240,11 @@ struct CastLikeInsertSliceOpFolder final
 
 void OptimizeTensorInsertExtractSlicesPass::runOnOperation() {
   auto funcOp = getOperation();
-  linalg::hoistRedundantVectorTransfers(cast<func::FuncOp>(funcOp));
   IRRewriter rewriter(funcOp->getContext());
+
+  funcOp.walk([&](scf::ForOp forOp) { moveLoopInvariantCode(forOp); });
+  LDBG("after hoisting loop invariant code\n" << funcOp);
```

Comment:
linalg::hoistRedundantVectorTransfers is a combination of LICM + memref optimizations. We only run LICM to keep the previous behavior for tensor code.